test/hotspot/jtreg/serviceability/sa/LingeredAppWithLock.java

```diff
@@ -36,24 +36,37 @@ public static void lockMethod(Object lock) {
         }
     }
 
+    public static void waitMethod(Object lock) {
+        synchronized (lock) {
+            try {
+                lock.wait(300000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
     public static void main(String args[]) {
         Thread classLock1 = new Thread(() -> lockMethod(LingeredAppWithLock.class));
         Thread classLock2 = new Thread(() -> lockMethod(LingeredAppWithLock.class));
         Thread objectLock = new Thread(() -> lockMethod(classLock1));
         Thread primitiveLock = new Thread(() -> lockMethod(int.class));
+        Thread objectWait = new Thread(() -> waitMethod(new Object()));
```

Comment:
Use of `new Object()` could be problematic here as the JIT can recognise that this is a non-escaping object and elide the synchronization.